### ISSUE TITLE
Crash when downloading Video from twitter

### ISSUE THREAD

**Issue by shipsywor** [bak840/twitter-video-downloader-android#1]
Error:

```
2021-04-20 02:54:02.787 12397-12397/com.bakulabs.twittervideodownloader E/DecorView[]: Destroying unexpected ActionMode instance of TYPE_FLOATING; com.android.internal.view.FloatingActionMode@49e7be1 was not the current floating action mode! Expected null
2021-04-20 02:54:06.687 12397-12397/com.bakulabs.twittervideodownloader E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.bakulabs.twittervideodownloader, PID: 12397
    java.lang.NullPointerException
        at com.bakulabs.twittervideodownloader.ui.home.HomeViewModel$downloadVariant$1.invokeSuspend(HomeViewModel.kt:113)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.internal.DispatchedContinuationKt.resumeCancellableWith(DispatchedContinuation.kt:349)
```

When this happens:

This happens when using twitter url like this: https://twitter.com/Huawei/status/1384114195348611082

BUT it does not happen for the twitter url of jujutsu kaisen that is shown by default

**Comment by bak840** [comment]
Thanks for telling me about that issue, it was because to download a variant, I use the tweet Id for the filename and I was too lazy to pass around the id from the tweet url in the method so I decided to get it from the variant url itself but not all variants url are constructed the same so the regex function worked on the jujutsu variants url but not this one. I did not know that. So I decided to store the url from the view in the viewmodel itself so that the downloadVariant method can access it and get the tweet Id. I've just commited the changes to the main branch.

Secondly , I did not expect someone to try this app lol and even less report an issue, it's my first issue, thanks, I did not even put a README yet. How did you manage for the Twitter API bearer token? Did you hardcoded it in the code? I use a properties file not commited to the repo and I was planning of explaining the procedure in the README.
